src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1802 @@
+{% endraw %}
+
+In JavaScript, PascalCase is the convention for classes and prototype constructors - essentially, anything that can have distinct instances. Vue components also have instances, so it makes sense to also use PascalCase. As an added benefit, using PascalCase within JSX (and templates) allows readers of the code to more easily distinguish between components and HTML elements.
+
+However, for applications that use **only** global component definitions via `Vue.component`, we recommend kebab-case instead. The reasons are:
+
+- It's rare that global components are ever referenced in JavaScript, so following a convention for JavaScript makes less sense.
+- These applications always include many in-DOM components, where kebab-case [**must** be used](#Component-name-casing-in-templates).
+{% raw %}</details>{% endraw %}
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` js
+Vue.component('myComponent', {
+  // ...
+})
+```
+
+``` js
+import myComponent from './MyComponent.vue'
+```
+
+``` js
+export default {
+  name: 'myComponent',
+  // ...
+}
+```
+
+``` js
+export default {
+  name: 'my-component',
+  // ...
+}
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+``` js
+Vue.component('MyComponent', {
+  // ...
+})
+```
+
+``` js
+Vue.component('my-component', {
+  // ...
+})
+```
+
+``` js
+import MyComponent from './MyComponent.vue'
+```
+
+``` js
+export default {
+  name: 'MyComponent',
+  // ...
+}
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Abbreviations in component names <sup data-p="b">strongly recommended</sup>
+
+**Component names should prefer full words over abbreviations.**
+
+The autocompletion in editors make the cost of writing longer names very low, while the clarity they provide is invaluable. Uncommon abbreviations, in particular, should always be avoided.
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+```
+components/
+|- SdSettings.vue
+|- UProfOpts.vue
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+```
+components/
+|- StudentDashboardSettings.vue
+|- UserProfileOptions.vue
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Prop name casing <sup data-p="b">strongly recommended</sup>
+
+**Prop names should always use camelCase during declaration, but kebab-case in templates and [JSX](render-function.html#JSX).**
+
+We're simply following the conventions of each language. Within JavaScript, camelCase is more natural. Within HTML, kebab-case is.
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` js
+props: {
+  'greeting-text': String
+}
+```
+
+``` html
+<WelcomeMessage greetingText="hi"/>
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+``` js
+props: {
+  greetingText: String
+}
+```
+
+``` html
+<WelcomeMessage greeting-text="hi"/>
```

Comment:
This example is a little strange for me because the tag name is PascalCase but attribute name is kebab-case. Maybe I prefer kebab-case tag name which is more natural for me. The second option is both PascalCases. The here is the 3rd. 😂

Reply:
Hmm, PascalCase is a convention used to identify instances in JavaScript. It applies to components, because they're really both JavaScript and instances. I think PascalCase would be strange for attributes, because they _don't_ have instances. Does that make sense?

Reply:
Ah, make sense. No problem. 👌